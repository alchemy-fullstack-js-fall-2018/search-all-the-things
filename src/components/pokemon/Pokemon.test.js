import React from 'react';
import Pokemons from './Pokemon';

describe('Pokemon test', () => {

  it('renders a list of Pokemon', () => {
    let pokemons = [];

    const pokemonComponents = pokemons.map(pokemon => {
      return <Pokemon key={pokemon.id}
      name={pokemon.name}/>;
    });

    const pokemonlist = new Pokemons({
      pokemonComponents: [
        'bulbasaur',
        'ivysaur',
        'venusaur',
        'charmander',
        'charmeleon',
        'charizard',
        'squirtle',
        'wartortle',
        'blastoise',
        'caterpie',
        'metapod',
        'butterfree',
        'pikachu',
        'raichu',
        'sandshrew',
        'sandslash',
        'nidoran-f',
        'nidorina'
      ]
    });
    expect(pokemonlist.render()).toMatchSnapshot();
  });

});
