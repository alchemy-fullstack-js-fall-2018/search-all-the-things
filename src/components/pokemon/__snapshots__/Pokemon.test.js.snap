// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon test renders a list of Pokemon 1`] = `
<React.Fragment>
  <div
    className="container"
  >
    <label
      className="label"
    >
      Select Abilities
    </label>
    <select
      defaultValue=""
      name="ability"
      onChange={[Function]}
    />
  </div>
  <div
    className="pageView"
  >
    <h3>
      Page view
    </h3>
    <Pageable
      currentPage={1}
      totalPages={1}
      updatePage={[Function]}
    />
  </div>
  <div>
    <h1>
      Pokemon List
    </h1>
    <ol />
  </div>
</React.Fragment>
`;
